pkg/library/libraryinputresources/resource.go

```diff
@@ -85,9 +85,21 @@ func ResourceFromFile(location, fileTrimPrefix string) (*Resource, error) {
 	retFilename := strings.TrimPrefix(location, fileTrimPrefix)
 	retFilename = strings.TrimPrefix(retFilename, "/")
 
+	// If the file represents a list, derive the resource name from the file name.
+	// Otherwise, if it represents a single resource, use the parent directory name instead.
+	resourceContent := ret.(*unstructured.Unstructured)
+	resourceName := filepath.Base(filepath.Dir(location))
+	if resourceContent.IsList() {
+		resourceName = strings.TrimSuffix(filepath.Base(location), ".yaml")
+	}
 	return &Resource{
+		ResourceType: schema.GroupVersionResource{
+			Group:    ret.GetObjectKind().GroupVersionKind().Group,
+			Version:  ret.GetObjectKind().GroupVersionKind().Version,
+			Resource: resourceName,
```

Comment:
does the resource have to be provided here ?
what is the location ?

Having a `GVK` and the discovery from a must-gather I think we would be able to figure out the `resource` part.

For example we could do something similar to https://github.com/openshift/library-go/blob/573ff142f0a0c98340ce1ae2553499b61dcbcb70/pkg/manifestclient/read_roundtripper.go#L194

Reply:
Agree on using discovery to determine resource. And also we should update this method to return a slice of these if consumers expect one instance per (likely)

Reply:
Could you give me a few pointers on how to get the resource information from discovery? The linked code retrieves discovery data from files provided with the manifestclient. I'm unsure if I should do the same and assume that these files will be available in input-dir, or if I should instead use a discovery client

Reply:
The input-dir must have must-gather like structure. Thus I would simply create a kube client or discovery client based on our manifest client and I would call the discovery endpoint for the given `GV` to get the resources

https://github.com/kubernetes/kubernetes/blob/462dd326c2e98d937a96d49002883000efe4b2d6/staging/src/k8s.io/client-go/discovery/discovery_client.go#L128